File: sdks/ruby/lib/daon/protection_result.rb
module Daon
  class ProtectionResult
    attr_reader :success, :tx_hash, :content_hash, :verification_url, :error, :timestamp

    def initialize(success:, content_hash:, tx_hash: nil, verification_url: nil, error: nil)
      @success = success
      @content_hash = content_hash
      @tx_hash = tx_hash
      @verification_url = verification_url
      @error = error
      @timestamp = Time.now
    end

    def protected?
      success
    end

    def failed?
      !success
    end

    def blockchain_url
      return nil unless tx_hash
      "https://explorer.daon.network/tx/#{tx_hash}"
    end

    def to_h
      {
        success: success,
        content_hash: content_hash,
        tx_hash: tx_hash,
        verification_url: verification_url,
        blockchain_url: blockchain_url,
        error: error,
        timestamp: timestamp.iso8601
      }
    end

    def to_json
      to_h.to_json
    end
  end
end

File: sdks/php/src/ProtectionResult.php
<?php

namespace Daon;

class ProtectionResult
{
    private bool $success;
    private string $contentHash;
    private ?string $txHash;
    private ?string $verificationUrl;
    private ?string $blockchainUrl;
    private ?string $error;
    private \DateTime $timestamp;

    public function __construct(array $data)
    {
        $this->success = $data['success'] ?? false;
        $this->contentHash = $data['content_hash'] ?? '';
        $this->txHash = $data['tx_hash'] ?? null;
        $this->verificationUrl = $data['verification_url'] ?? null;
        $this->blockchainUrl = $data['blockchain_url'] ?? null;
        $this->error = $data['error'] ?? null;
        $this->timestamp = $data['timestamp'] ?? new \DateTime();
    }

    public function isSuccess(): bool { return $this->success; }
    public function getContentHash(): string { return $this->contentHash; }
    public function getTxHash(): ?string { return $this->txHash; }
    public function getVerificationUrl(): ?string { return $this->verificationUrl; }
    public function getBlockchainUrl(): ?string { return $this->blockchainUrl; }
    public function getError(): ?string { return $this->error; }
    public function getTimestamp(): \DateTime { return $this->timestamp; }
}
